subsys/bluetooth/controller/ll_sw/ctrl.c

```diff
@@ -1317,15 +1339,37 @@ static inline u32_t isr_rx_scan(u8_t devmatch_ok, u8_t devmatch_id,
 		    (struct radio_le_conn_cmplt *)&pdu_data->payload;
 		radio_le_conn_cmplt->status = 0x00;
 		radio_le_conn_cmplt->role = 0x00;
-		radio_le_conn_cmplt->peer_addr_type = pdu_adv_tx->rx_addr;
-		memcpy(&radio_le_conn_cmplt->peer_addr[0],
-		       &pdu_adv_tx->payload.connect_ind.adv_addr[0],
-		       BDADDR_SIZE);
+#if defined(CONFIG_BLUETOOTH_CONTROLLER_PRIVACY)
 		radio_le_conn_cmplt->own_addr_type = pdu_adv_tx->tx_addr;
 		memcpy(&radio_le_conn_cmplt->own_addr[0],
 		       &pdu_adv_tx->payload.connect_ind.init_addr[0],
 		       BDADDR_SIZE);
-		radio_le_conn_cmplt->peer_irk_index = irkmatch_id;
+
+		if (irkmatch_ok && rl_idx != FILTER_IDX_NONE) {
+			/* Store identity address */
+			ctrl_id_addr_get(rl_idx,
```

Comment:
I have a patch that will do this, but I want to make sure that this is actually safe. In the meantime, keep the copy in ISR